metricflow/naming/object_builder_scheme.py

```diff
@@ -0,0 +1,200 @@
+                    fields_to_compare=tuple(fields_to_compare),
+                )
+            )
+
+        for entity_call_parameter_set in call_parameter_sets.entity_call_parameter_sets:
+            return EntityLinkPattern(
+                EntityLinkPatternParameterSet(
+                    element_name=entity_call_parameter_set.entity_reference.element_name,
+                    entity_links=entity_call_parameter_set.entity_path,
+                    time_granularity=None,
+                    date_part=None,
+                    fields_to_compare=(
+                        ParameterSetField.ELEMENT_NAME,
+                        ParameterSetField.ENTITY_LINKS,
+                    ),
+                )
+            )
+
+        raise RuntimeError("There should have been a return associated with one of the CallParameterSets.")
+
+    @override
+    def input_str_follows_scheme(self, input_str: str) -> bool:
+        if ObjectBuilderNamingScheme._NAME_REGEX.match(input_str) is None:
+            return False
+        try:
+            call_parameter_sets = WhereFilterParser.parse_call_parameter_sets("{{ " + input_str + " }}")
+            return_value = (
+                len(call_parameter_sets.dimension_call_parameter_sets)
+                + len(call_parameter_sets.time_dimension_call_parameter_sets)
+                + len(call_parameter_sets.entity_call_parameter_sets)
+            ) == 1
+            return return_value
+        except ParseWhereFilterException:
+            return False
+
+    @override
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(id()={hex(id(self))})"
+
+
+class _ObjectBuilderNameTransform(LinkableSpecSetTransform[Sequence[str]]):
+    """Transforms specs into strings following the object builder scheme."""
+
+    @staticmethod
+    def _get_initializer_parameter_str(
+        element_name: str,
+        entity_links: Sequence[EntityReference],
+        time_granularity: Optional[TimeGranularity],
+        date_part: Optional[DatePart],
+    ) -> str:
+        """Return the parameters that should go in the initializer.
+
+        e.g. `'user__country', time_granularity_name='month'`
+        """
+        initializer_parameters = []
+        entity_link_names = list(entity_link.element_name for entity_link in entity_links)
+        if len(entity_link_names) > 0:
+            initializer_parameters.append(repr(entity_link_names[-1] + DUNDER + element_name))
+        else:
+            initializer_parameters.append(repr(element_name))
+        if time_granularity is not None:
+            initializer_parameters.append(
+                f"'{time_granularity.value}'",
+            )
+        if date_part is not None:
+            initializer_parameters.append(f"date_part_name={repr(date_part.value)}")
+        if len(entity_link_names) > 1:
+            initializer_parameters.append(f"entity_path={repr(entity_link_names[:-1])}")
+
+        return ", ".join(initializer_parameters)
```

Comment:
I feel like there has to be a better way to do this but without overhauling our spec property interfaces I don't know what it is. The logic here is a little brittle because of the assumption that all the parameters passed are correct for the spec type but since it's all in this one small class I think it's fine.